docs/quickstarts/outlook-quickstart-json-manifest.md

```diff
@@ -0,0 +1,257 @@
+
+1. Choose the **Show Taskpane** button in the ribbon to open the add-in task pane.
+
+    > [!NOTE]
+    > If you receive the error "We can't open this add-in from localhost" in the task pane, follow the steps outlined in the [troubleshooting article](/office/troubleshoot/office-suite-issues/cannot-open-add-in-from-localhost).
+
+1. When prompted with the **WebView Stop On Load** dialog box, select **OK**.
+
+    [!INCLUDE [Cancelling the WebView Stop On Load dialog box](../includes/webview-stop-on-load-cancel-dialog.md)]
+
+1. Scroll to the bottom of the task pane and choose the **Run** link to copy the message's subject to the task pane.
+
+1. End the debugging session with the following command:
+
+    ```command&nbsp;line
+    npm stop
+    ```
+
+    > [!IMPORTANT]
+    > Closing the web server window does not reliably shut down the web server. If it is not properly shut down, you'll encounter problems as you change and rerun the project.
+
+1. Close all instances of Outlook.
+
+## Add a custom button to the ribbon
+
+Let's add a custom button to the ribbon that inserts text into a message body.
+
+1. Open your project in VS Code or your preferred code editor.
+   [!INCLUDE [Instructions for opening add-in project in VS Code via command line](../includes/vs-code-open-project-via-command-line.md)]
+
+1. In your code editor, open the file **./src/command/command.ts** and add the following code to the end of the file. This function will insert "Hello World" at the cursor point in message body.
+
+    ```js
+    function insertHelloWorld(event: Office.AddinCommands.Event) {
+        Office.context.mailbox.item.body.setSelectedDataAsync("Hello World", {coercionType: Office.CoercionType.Text});
+
+        // Be sure to indicate when the add-in command function is complete
+        event.completed();
+    }
+
+    // Put the function on the global namespace
+    g.insertHelloWorld = insertHelloWorld;
+    ```
+
+1. Open the file **./manifest/manifest.json**.
+
+    > [!NOTE]
+    > When referring to nested JSON properties, this article uses dot notation. When an item in an array is referenced, the bracketed zero-based number of the item is used. 
+
+1. To write to a message, the add-in's permissions need to be raised. Scroll to the property `authorization.permissions.resourceSpecific[0].name` and change the value to "MailboxItem.ReadWrite.User".
+
+1. When an add-in command executes code instead of opening a task pane, it must run the code in a JavaScript runtime that is separate from the embedded webview in which task pane code runs. So the manifest must specify an additional runtime. Scroll to the property `extension.runtimes` and add the following object to the `runtimes` array. Be sure to put a comma after the object that is already in the array. Note the following about this markup:
```

Comment:
Avoid "executes" for localization purposes.
```suggestion
1. When an add-in command runs code instead of opening a task pane, it must run the code in a JavaScript runtime that is separate from the embedded webview in which task pane code runs. So the manifest must specify an additional runtime. Scroll to the property `extension.runtimes` and add the following object to the `runtimes` array. Be sure to put a comma after the object that is already in the array. Note the following about this markup:
```